make-rules/ips.mk

```diff
@@ -412,32 +417,44 @@ FRC:
 PKGSEND_PUBLISH_OPTIONS = -s $(WS_REPO) publish --fmri-in-manifest
 PKGSEND_PUBLISH_OPTIONS += $(PKG_PROTO_DIRS:%=-d %)
 PKGSEND_PUBLISH_OPTIONS += -T \*.py
-$(MANIFEST_BASE)-%.published:	$(MANIFEST_BASE)-%.depend.res $(BUILD_DIR)/.linted-$(MACH)
+
+# Do all the hard work that is needed to ensure the package is consistent
+# and ready for publishing, except actually pushing bits to a repository
+$(MANIFEST_BASE)-%.pre-published:	$(MANIFEST_BASE)-%.depend.res $(BUILD_DIR)/.linted-$(MACH)
+	$(CP) $< $@
+	@echo "NEW PACKAGE CONTENTS ARE LOCALLY VALIDATED AND READY TO GO"
+
+# Push to the repo
+$(MANIFEST_BASE)-%.published:	$(MANIFEST_BASE)-%.pre-published
+	@echo "PUSHING NEW PACKAGE CONTENTS TO THE REPO"
```

Comment:
Following suit of other actions in this file, many of which are quite
verbose, and to warn about a potentially long operation (e.g. remote repo)
or one that user wants aborted (Ctrl+C) at this time.

From what I gathered on IRC discussion, reaction was "Cool, let's do more
of those" - but maybe that was sarcasm? ;)

2016-06-11 23:06 GMT+02:00 Adam Števko <notifications@github.com>:

> In make-rules/ips.mk
> <https://github.com/OpenIndiana/oi-userland/pull/2047#discussion_r66711726>
> :
>
> > @@ -412,32 +417,44 @@ FRC:
> >  PKGSEND_PUBLISH_OPTIONS = -s $(WS_REPO) publish --fmri-in-manifest
> >  PKGSEND_PUBLISH_OPTIONS += $(PKG_PROTO_DIRS:%=-d %)
> >  PKGSEND_PUBLISH_OPTIONS += -T \*.py
> > -$(MANIFEST_BASE)-%.published:	$(MANIFEST_BASE)-%.depend.res $(BUILD_DIR)/.linted-$(MACH)
> > +
> > +# Do all the hard work that is needed to ensure the package is consistent
> > +# and ready for publishing, except actually pushing bits to a repository
> > +$(MANIFEST_BASE)-%.pre-published:	$(MANIFEST_BASE)-%.depend.res $(BUILD_DIR)/.linted-$(MACH)
> > +	$(CP) $< $@
> > +	@echo "NEW PACKAGE CONTENTS ARE LOCALLY VALIDATED AND READY TO GO"
> > +
> > +# Push to the repo
> > +$(MANIFEST_BASE)-%.published:	$(MANIFEST_BASE)-%.pre-published
> > +	@echo "PUSHING NEW PACKAGE CONTENTS TO THE REPO"
>
> Why are you changing the behaviour by adding text?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenIndiana/oi-userland/pull/2047/files/4221e17e0bba8bf0ee49ce857a6d61200767cbf7#r66711726>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABj5lJy-uvXQcl1XPSN4iKwdYtRhBrZkks5qKyNVgaJpZM4Ina49>
> .
>